src/lib/comptime/gen.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+if [[ -n "${DUNE_PROFILE:-}" ]]; then
+    dune_profile_val="Some \"$DUNE_PROFILE\""
+else
+    dune_profile_val="None"
+fi
+
+printf 'let dune_profile : string option = %s\n' "$dune_profile_val" > "$1"
```

Comment:
This is the ugliest part of this solution. We could potentially use https://github.com/stedolan/ppx_stage. But that lib is too old and we need to have it update to date. 

Right now I guess this hack is fine, as it's already used in `mina_version.normal`

Reply:
> This is the ugliest part of this solution.

Is it really that ugly? :). I like it 😄

Reply:
Code generation that ignores AST structure is not ideal, I'd say. We'll get rid of it in later PRs though